applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -0,0 +1,741 @@
+			err = cloud_disconnect(nrf_cloud);
+			if (err) {
+				LOG_ERR("Cloud disconnection failed, error: %d", err);
+			}
+			nrf_cloud_ready = false;
+			err = nrf_modem_gnss_stop();
+		} else {
+			err = -EINVAL;
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XPGPS: (%d,%d),<interval>,<timeout>\r\n",
+			PGPS_STOP, PGPS_START);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
+/**@brief handle AT#XCELLPOS commands
+ *  AT#XCELLPOS=<op>
+ *  AT#XCELLPOS? READ command not supported
+ *  AT#XCELLPOS=?
+ */
+int handle_at_cellpos(enum at_cmd_type cmd_type)
+{
+	int err = -EINVAL;
+	uint16_t op;
+
+	switch (cmd_type) {
+	case AT_CMD_TYPE_SET_COMMAND:
+		err = at_params_unsigned_short_get(&at_param_list, 1, &op);
+		if (err < 0) {
+			return err;
+		}
+		if (op == CELLPOS_START_SCELL || op == CELLPOS_START_MCELL) {
+			if (op == CELLPOS_START_SCELL) {
+				cell_pos_type = CELL_POS_TYPE_SINGLE;
+			} else {
+				sprintf(rsp_buf, "\r\n#XCELLPOS: \"not supported\"\r\n");
+				rsp_send(rsp_buf, strlen(rsp_buf));
+				return -ENOTSUP;
+			}
+			run_type = RUN_TYPE_CELL_POS;
+			if (!nrf_cloud_ready) {
+				err = cloud_connect(nrf_cloud);
+				if (err) {
+					LOG_ERR("Cloud connection failed, error: %d", err);
+				}
+#if defined(NRF_CLOUD_ACCESS)
+			} else {
+				on_cloud_evt_ready();
+#endif
+			}
+		} else if (op == CELLPOS_STOP) {
+			err = cloud_disconnect(nrf_cloud);
+			if (err) {
+				LOG_ERR("Cloud disconnection failed, error: %d", err);
+			}
+			nrf_cloud_ready = false;
+		} else {
+			err = -EINVAL;
+		} break;
+
+	case AT_CMD_TYPE_TEST_COMMAND:
+		sprintf(rsp_buf, "\r\n#XCELLPOS: (%d,%d,%d)\r\n",
+			CELLPOS_STOP, CELLPOS_START_SCELL, CELLPOS_START_MCELL);
+		rsp_send(rsp_buf, strlen(rsp_buf));
+		err = 0;
+		break;
+
+	default:
+		break;
+	}
+
+	return err;
+}
+
+/**@brief API to initialize GNSS AT commands handler
+ */
+int slm_at_gnss_init(void)
+{
+	int err = 0;
+
+	err = nrf_modem_gnss_init();
+	if (err) {
+		LOG_ERR("Could not initialize GNSS, error: %d", err);
+		return err;
+	}
+	err = nrf_modem_gnss_event_handler_set(gnss_event_handler);
+	if (err) {
+		LOG_ERR("Could set GNSS event handler, error: %d", err);
+		return err;
+	}
+
+#if defined(NRF_CLOUD_ACCESS)
+	if (nrf_cloud == NULL) {
+		nrf_cloud = cloud_get_binding("NRF_CLOUD");
+		if (nrf_cloud == NULL) {
+			LOG_ERR("Could not get binding to backend");
+			return -EINVAL;
+		}
+		err = cloud_init(nrf_cloud, cloud_event_handler);
+		if (err) {
+			LOG_ERR("Cloud backend could not be initialized, error: %d", err);
+			return err;
+		}
+	}
+#endif
+
+	k_work_init(&agps_req, agps_req_wk);
+	k_work_init(&pgps_req, pgps_req_wk);
+	k_work_init(&cell_pos_req, cell_pos_req_wk);
+
+	return err;
+}
+
+/**@brief API to uninitialize GNSS AT commands handler
+ */
+int slm_at_gnss_uninit(void)
+{
+#if defined(NRF_CLOUD_ACCESS)
+	(void)cloud_disconnect(nrf_cloud);
+	(void)cloud_uninit(nrf_cloud);
```

Comment:
this is not implemented for nrf_cloud.
i created a [ticket](https://projecttools.nordicsemi.no/jira/browse/CIA-346) to implement un-init.

Reply:
un-init is now implemented in nrf_cloud lib :tada: